src/components/chips/js/chipsController.js

```diff
@@ -50,6 +51,26 @@ function MdChipsCtrl ($scope, $mdConstant, $log, $element, $timeout) {
   /** @type {boolean} */
   this.hasAutocomplete = false;
 
+  /** @type {string} */
+  this.$scope.enableChipEdit = $attrs['mdEnableChipEdit'] || '';
+
+  /** @type {function} */
+  this.$scope.updateChipContents = function(chipIndex, chipContents){
+    if(chipIndex >= 0 && chipIndex < this.items.length) {
+      this.items[chipIndex] = chipContents;
+      this.ngModelCtrl.$setDirty();
+    }
+  }.bind(this);
+
+  /** @type {function} */
+  this.$scope.removeChipAtIndex = function(chipIndex){
+    this.removeChip(chipIndex);
+  }.bind(this);
+
+  /** @type {function} */
+  this.$scope.getSelectedChipIndex = function(){
```

Comment:
When using the `injected` controller instead of the `$scope.$parent` (as above said), then you should append these methods to the prototype.